pkg/api/composition.go

```diff
@@ -45,11 +45,21 @@ type Global struct {
 	// BuildConfig specifies the build configuration for this run.
 	BuildConfig map[string]interface{} `toml:"build_config" json:"build_config"`
 
+	// Build applies global build defaults that trickle down to all groups, such
+	// as selectors or dependencies. Groups can override these in their local
+	// build definition.
+	Build *Build `toml:"build" json:"build"`
+
 	// Runner is the runner we're using.
 	Runner string `toml:"runner" json:"runner" validate:"required"`
 
 	// RunConfig specifies the run configuration for this run.
 	RunConfig map[string]interface{} `toml:"run_config" json:"run_config"`
+
+	// Run applies global run defaults that trickle down to all groups, such as
+	// test parameters or build artifacts. Groups can override these in their
+	// local run definition.
+	Run *Run `toml:"run" json:"run"`
```

Comment:
I wonder if we can come up with slightly better names here, as it is starting to get confusing - Build and BuildConfig, Run and RunConfig...

Maybe:

```
Build -> GroupBuildParams
Run -> GroupRunParams
```

The idea is that Build and Run are applied to the Groups, whereas BuildConfig and RunConfig are global for the given build and run.

In any case that can be done as a follow up or not at all.

Reply:
Yup, we need to deduplicate between build, builder, run, runner, parameters, config...